receiver/azuremonitorreceiver/scraper_batch.go

```diff
@@ -359,6 +359,7 @@ func (s *azureBatchScraper) getResourceMetricsDefinitionsByType(ctx context.Cont
 			timeGrain := *v.MetricAvailabilities[0].TimeGrain
 			dimensions := filterDimensions(v.Dimensions, s.cfg.Dimensions, resourceType, metricName)
 			compositeKey := metricsCompositeKey{
+				namespace:    *v.Namespace,
```

Comment:
I tested locally and this is actually a part of the fix. At "list metrics def" query time a bit upper, I need to precise each namespace instead of nil options. 

I don't think there is a way to discovery automatically all the namespaces. So user will have to choose between default namespace (nil) or list explicitely each namespace.

Reply:
This is actually possible to get all namespaces thanks to the ``NewMetricNamespacesClient``. Then filter should be possible yes but also an auto discovery.